bentoml/_internal/frameworks/pytorch.py

```diff
@@ -86,13 +86,27 @@ def load(
                 f"Model {tag} was saved with module {bentoml_model.info.module}, failed loading with {MODULE_NAME}."
             )
     weight_file = bentoml_model.path_of(f"{SAVE_NAMESPACE}{PT_EXT}")
-    # TorchScript Models are saved as zip files
-    if zipfile.is_zipfile(weight_file):
+    model_format = bentoml_model.info.options.get("_model_format")
+    # backward compatibility
+    if not model_format:
+        if zipfile.is_zipfile(weight_file):
+            model_format = "torchscript:v1"
+        else:
+            model_format = "cloudpickle:v1"
+
+    if model_format == "torchscript:v1":
         model: "torch.jit.ScriptModule" = torch.jit.load(weight_file, map_location=device_id)  # type: ignore[reportPrivateImportUsage] # noqa: LN001
-        return model
-    else:
+    elif model_format == "cloudpickle:v1":
         with Path(weight_file).open("rb") as file:
-            return cloudpickle.load(file)
+            model: "torch.nn.Module" = cloudpickle.load(file).to(device_id)
+    elif model_format == "torch.save:v1":
+        with Path(weight_file).open("rb") as file:
+            model: "torch.nn.Module" = torch.load(file, map_location=device_id)
```

Comment:
we cannot use `torch.load(file, pickle_module=cloudpickle` because `cloudpickle` does not implement `Unpickler` class. However data (including class/lambda/function) pickled by `cloudpickle` should be unpickled by `pickle` module so this line still works

Reply:
Can we adapt the cloudpickle to support Unpickler?